android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -140,21 +149,35 @@ public String getCarrier() {
   }
 
   @ReactMethod
-  public Integer getTotalDiskCapacity() {
+  public Long getTotalDiskCapacity() {
     try {
-      StatFs root = new StatFs(Environment.getRootDirectory().getAbsolutePath());
-      return root.getBlockCount() * root.getBlockSize();
+      long totalSize=0;
+      if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT){
+        File[] files = this.reactContext.getExternalFilesDirs(Environment.MEDIA_MOUNTED);
+        for (File file : files) {
+          totalSize += this.getCapacity(file);
+        }
+      }else {
+        totalSize += this.getCapacity(Environment.getExternalStorageDirectory());
+      }
+      totalSize += this.getCapacity(Environment.getDownloadCacheDirectory());
+      totalSize += this.getCapacity(Environment.getRootDirectory());
+      totalSize += this.getCapacity(Environment.getDataDirectory());
```

Comment:
Tested this code with my OnePlus5T (64G internal memory, no external memory) this returns `172 612 931 584`, which is incorrect.

My Android storage screen reports 56.55GB.

From Android docs for Environment.getExternalStorageDirectory():
> Note: don't be confused by the word "external" here. This directory can better be thought as media/shared storage. It is a filesystem that can hold a relatively large amount of data and that is shared across all applications (does not enforce permissions). Traditionally this is an SD card, but it may also be implemented as built-in storage in a device that is distinct from the protected internal storage and can be mounted as a filesystem on a computer.

So adding external storage to download & root & data is not always the correct way to go as they are sometime the same. Do you think that we could add a test for the sd card presence and act accordingly?